cdap-api/src/main/java/co/cask/cdap/api/schedule/SchedulableProgramType.java

```diff
@@ -20,5 +20,5 @@
  * Defines types of programs that can be scheduled by the Application.
  */
 public enum SchedulableProgramType {
-  WORKFLOW, MAPREDUCE, SPARK, CUSTOM_ACTION
+  WORKFLOW, MAPREDUCE, SPARK, CUSTOM_ACTION, NONE
```

Comment:
Yes. Thats true. The implementation can be modified as - `WorkflowNode` can have `name` and `WorkflowNodeType`. Maintain the `map<String, ScheduleProgramInfo> programs` in the `WorkflowSpecification`. If `WorkflowNode` is of type `ACTION`, then corresponding `ScheduleProgramInfo` can be fetched from the `programs` map using `name`. Please let me know if it is more clear.